src/main.cpp

```diff
@@ -2594,201 +2595,255 @@ void vcMain_RenderSceneWindow(vcState *pProgramState)
     vcCamera_UpdateMatrices(pProgramState->geozone, &pProgramState->pActiveViewport->camera, pProgramState->settings.camera, udFloat2::create(pProgramState->pActiveViewport->resolution));
   }
 
-  if (vcHotkey::IsPressed(vcB_Fullscreen) || ImGui::IsNavInputTest(ImGuiNavInput_TweakFast, ImGuiInputReadMode_Released))
-    vcMain_PresentationMode(pProgramState);
-
+  udDouble3 cameraMoveOffset = udDouble3::zero();
   vcRenderSceneUI(pProgramState, windowPos, windowSize, &cameraMoveOffset);
 
   {
-    vcRender_BeginFrame(pProgramState, pProgramState->pActiveViewport->pRenderContext, renderData);
+    ImGui::PushStyleVar(ImGuiStyleVar_ItemSpacing, ImVec2(0, 0));
+    ImGui::Columns(pProgramState->activeViewportCount);
 
-    // Actual rendering to this texture is deferred
-    ImGui::Image(renderData.pSceneTexture, windowSize, ImVec2(0, 0), ImVec2(renderData.sceneScaling.x, renderData.sceneScaling.y));
+    // At the moment some functionality is restricted to certain viewports
+    for (int viewportIndex = 0; viewportIndex < pProgramState->activeViewportCount; ++viewportIndex)
+    {
+      ImVec2 viewportportPos = ImVec2(windowPos.x + ImGui::GetCursorPosX(), windowPos.y + ImGui::GetCursorPosY());
+      udFloat2 viewportportResolution = udFloat2::create(ImGui::GetColumnWidth(), windowSize.y);
 
-    if (pProgramState->settings.screenshot.taking)
-      pProgramState->screenshot.pImage = renderData.pSceneTexture;
+      vcRenderData renderData = {};
 
-    static bool wasContextMenuOpenLastFrame = false;
-    bool useTool = (io.MouseDragMaxDistanceSqr[0] < (io.MouseDragThreshold * io.MouseDragThreshold)) && ImGui::IsMouseReleased(0) && ImGui::IsItemHovered();
+      renderData.models.Init(32);
+      renderData.fences.Init(32);
+      renderData.labels.Init(32);
+      renderData.waterVolumes.Init(32);
+      renderData.polyModels.Init(64);
+      renderData.images.Init(32);
+      renderData.lines.Init(32);
+      renderData.viewSheds.Init(32);
+      renderData.pins.Init(512);
+      renderData.mouse.position.x = (uint32_t)(io.MousePos.x - viewportportPos.x);
+      renderData.mouse.position.y = (uint32_t)(io.MousePos.y - viewportportPos.y);
+      renderData.mouse.clicked = io.MouseClicked[1];
 
-    if ((io.MouseDragMaxDistanceSqr[1] < (io.MouseDragThreshold * io.MouseDragThreshold) && ImGui::BeginPopupContextItem("SceneContext")))
-    {
-      static bool hadMouse = false;
-      static udDouble3 mousePosCartesian;
-      static udDouble3 mousePosLongLat;
+      pProgramState->pActiveViewport = &pProgramState->pViewports[viewportIndex];
 
-      if (!wasContextMenuOpenLastFrame || ImGui::IsMouseClicked(1))
+      // TODO: Screenshot which viewport?
```

Comment:
Both as a single image - if the user wants a screenshot of just one, they should close the other viewport.

Reply:
Sounds reasonable.